GameOfRunes/GameOfRunes/Levels and Waves/LevelCreator.swift

```diff
@@ -0,0 +1,259 @@
+        return levelSpawnInterval
+    }
+
+    private static func getLevelDataAndSpawnInterval(levelNumber: Int) throws -> (EnemySpawnUnit, Double) {
+        switch levelNumber {
+        case -1:
+            return testLevel
+        case 1:
+            return level1
+        case 2:
+            return level2
+        case 3:
+            return level3
+        default:
+            throw LevelWaveError.invalidLevelNumber
+        }
+    }
+
+    static var level1: (EnemySpawnUnit, Double) {
+        do {
+            let basicOrcUnit = try EnemySpawnUnit(.orc1)
+            let basicTrollUnit = try EnemySpawnUnit(.troll1)
+            let alternatingUnit = basicOrcUnit + basicTrollUnit
+            let fullLevel = alternatingUnit + basicOrcUnit
+            let levelSpawnInterval = 3.0
+            return (fullLevel, levelSpawnInterval)
+        } catch {
+            fatalError("Unexpected error encountered: \(error)")
+        }
+    }
+
+    static var level2: (EnemySpawnUnit, Double) {
+        do {
+            let basicOrcUnit = try EnemySpawnUnit(.orc1, .orc2)
+            let basicKnightUnit = try EnemySpawnUnit(.evilKnight, .none, .evilKnight)
+            let alternatingUnit = basicOrcUnit + basicKnightUnit
+            let fullLevel = alternatingUnit + alternatingUnit
+            let levelSpawnInterval = 2.0
+            return (fullLevel, levelSpawnInterval)
+        } catch {
+            fatalError("Unexpected error encountered: \(error)")
+        }
+    }
+
+    static var level3: (EnemySpawnUnit, Double) {
+        do {
+            let basicKnightUnit = try EnemySpawnUnit(.evilKnight, .evilKnight, .evilKnight)
+            let basicOrcUnit = try EnemySpawnUnit(.orc1, .orc2, .orc1)
+            let alternatingUnit = basicKnightUnit + basicOrcUnit
+            let fullLevel = alternatingUnit + alternatingUnit + alternatingUnit
+            let levelSpawnInterval = 1.0
+            return (fullLevel, levelSpawnInterval)
+        } catch {
+            fatalError("Unexpected error encountered: \(error)")
+        }
+    }
+
+    static var testLevel: (EnemySpawnUnit, Double) {
+        do {
+            let fullLevel = try EnemySpawnUnit(.orc1)
+            let levelSpawnInterval = 1.0
+            return (fullLevel, levelSpawnInterval)
+        } catch {
+            fatalError("Unexpected error encountered: \(error)")
+        }
+    }
+}
+
+/** Extension for automatically creating levels based on desired difficulty */
+extension LevelCreator {
+    private static var monsterToDifficultyMap: [EnemyType: Int] = [
+        .evilKnight: GameConfig.Enemy.evilKnightDifficulty,
+        .orc1: GameConfig.Enemy.orc1Difficulty,
+        .orc2: GameConfig.Enemy.orc2Difficulty,
+        .orc3: GameConfig.Enemy.orc3Difficulty,
+        .troll1: GameConfig.Enemy.troll1Difficulty,
+        .troll2: GameConfig.Enemy.troll2Difficulty,
+        .troll3: GameConfig.Enemy.troll3Difficulty,
+        .none: 0
+    ]
```

Comment:
Put this as a computer property in `EnemyType` enum.

Reply:
Yup u r right. Dk why i didn't do that lol. Refactored!